src/snovault/elasticsearch/simple_queue.py

```diff
@@ -0,0 +1,156 @@
+'''
+Simple uuid queue for indexing process
+- Can be overridden by another script, class, module as long as it exposes
+the proper functions used in the indexer classes.
+'''
+import time
+from os import getpid
+
+
+class SimpleUuidServer(object):
+    '''Simple uuid queue as a list'''
+    def __init__(self):
+        self._uuids = []
+        self._client_conns = {}
+        self.errors = []
+
+    def _get_uuid(self):
+        if self._uuids:
+            return self._uuids.pop()
+        return None
+
+    def _load_uuid(self, uuid):
+        '''
+        Load uuid into queue
+        - Uuid ordering impacts indexing time, list is reversed due to
+        pop being used in get uuid
+        '''
+        self._uuids.insert(0, uuid)
+
+    def get_uuids(self, client_id, cnt):
+        '''The only way to get uuids from queue'''
+        uuids = []
+        if client_id not in self._client_conns:
+            self._client_conns[client_id] = {
+                'uuid_cnt': 0,
+                'get_cnt': 0,
+                'results': None,
+            }
+        client_conn = self._client_conns[client_id]
+        client_conn['get_cnt'] += 1
+        if cnt == -1:
+            cnt = len(self._uuids)
+        if cnt and not client_conn['uuid_cnt']:
+            # There are uuids on the server and the worker is not working
+            while cnt > 0:
+                uuid = self._get_uuid()
+                if not uuid:
+                    break
+                uuids.append(uuid)
+                cnt -= 1
+            client_conn['uuid_cnt'] = len(uuids)
+        return uuids
+
+    def has_uuids(self):
+        '''Are there uuids in the queue'''
+        if self._uuids:
```

Comment:
how about:

return  self._uuids

Reply:
That would actually return the uuids.  Here we just want a way to verify there are uuids.  

Other queue types have to make requests to get queue info and have different ways to check the queue length.   AWS SQS queue won't actually return the exact queue len for example.  It estimates in a way.  And redis has different function call depending on the queue type, set or list...there are a few others.  

In the end we just want the has_uuids function to return True or False if there are uuids remaining.